src/modules/systemlib/param/param.c

```diff
@@ -127,6 +127,7 @@ get_param_info_count(void)
 {
 	/* Singleton creation of and array of bits to track changed values */
 	if (!param_changed_storage) {
+		/* Note that we have a (highly unlikely) race condition here: in the worst case the allocation is done twice */
```

Comment:
@bkueng 

Would you be able to avoid the param_init() and uses this singleton to init the semaphore.  Also wrap this in a schedule_lock to avoid the race